utils/build_profiler_report.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/python3
+import sys
+
+
+def LoadLogFile(file) -> dict:
+    """Loads Targets and their durations from ninja logfile `file` and returns them in a dict"""
+
+    with open(file, 'r') as fp:
+        lines = fp.read().splitlines()
+
+    # {Target: duration} dict
+    TargetDurationDict = {}
+    for line in lines:
+        line = line.split()
+
+        # ToDo - create better comment detection
+        if line[0][0] != '#':
+
+            # calculate target compilation duration and add it to dict
+            duration = int(line[1]) - int(line[0])
+            TargetDurationDict[line[3]] = duration
+
+    return TargetDurationDict
+
+
+def FormatTime(t):
+    """Converts a time into a human-readable format"""
+
+    t /= 1000
+    if t < 60:
+        return '%.1fs' % t
+    if t < 60 * 60:
+        return '%dm%.1fs' % (t / 60, t % 60)
+    return '%dh%dm%.1fs' % (t / (60 * 60), t % (60 * 60) / 60, t % 60)
+
+
+def GenerateWebTreeMap(CurrentDict: dict, BaselineDict: dict) -> None:
+    """Create file for webtreemap to generate an HTML from; if target is new, append _NEW"""
+    with open(sys.argv[1] + ".webtreemap", 'w') as fp:
```

Comment:
Don't force your function to get input from parsing arguments, delegate that to another layer and let that function to accept a string as an input.